util/chplenv/printchplbuilds.py

```diff
@@ -0,0 +1,449 @@
+    for line in lines:
+        if len(line) == 0:
+            continue
+        (var, value) = line.split('=', 1)
+        if var in ['CHPL_HOME']:
+            continue
+        config[var] = value
+        if var == 'CHPL_COMM' and value != "none":
+            # CHPL_COMM_DEBUG isn't returned by printchplenv
+            value = '+' if os.getenv('CHPL_COMM_DEBUG') else '-'
+            config['CHPL_COMM_DEBUG'] = value
+    return config
+
+# Returns a list of variables that differ between two configurations 
+
+def FindDiffVars(usedvars, a, b):
+    result = []
+    for var in usedvars:
+        value = a.get(var, 'NA')
+        if a.get(var, 'NA') != b.get(var, 'NA'):
+            result.append(var)
+    return result
+
+# print shell commands to change the environment variables from current to config
+
+def PrintShellCommands(current, config, shell="bash"):
+    global usedvars
+    if shell == "csh":
+        print("set noglob;")
+    for var in usedvars:
+        newValue = config.get(var, None)
+        oldValue = current.get(var, None)
+        if oldValue != newValue:
+            if newValue is None or newValue == '-':
+                if shell == "csh":
+                    print('unsetenv %s;' % (var))
+                elif shell == "bash":
+                    print('unset %s;' % (var))
+            else:
+                if newValue == '+':
+                    newValue = 1
+                if shell == "csh":
+                    print('setenv %s "%s";' %(var, newValue))
+                elif shell == "bash":
+                    print('export %s="%s";' %(var, newValue))
+    if shell == "csh":
+        print("unset noglob;")
+
+# Prints the configurations in columns. If a variable's value differs from the
+# current configuration it has a '*' suffix. "extras" are variables that shouldn't
+# get a prefix, e.g. MTIME.
+
+def Output(current, variables, extras, configs):
+    for var in variables + extras:
+        output = "%25s:" % var
+        for config in configs:
+            value = config.get(var, 'NA')
+            if var not in extras:
+                tag = "" if current.get(var, 'NA') == value else "*"
+                value += tag
+            if var == 'MTIME' and value != 'NA':
+                value = datetime.datetime.fromtimestamp(value).strftime("%b %d %H:%M")
+            output += " %-20s" % value
+        print(output)
+
+def main(args):
+    global configs
+    global used
+    global usedvars
+    usage = "Usage %prog [options]"
+    description = "Displays the available Chapel runtime builds. Values that differ from" \
+    " the current configuration have a '*' suffix. '+' denotes a variable that is set but" \
+    " whose value doesn't matter, '-' denotes a variable that is not set. 'NA' denotes a" \
+    " variable that is not applicable to the build." 
+
+    parser = optparse.OptionParser(version = "1.0", usage=usage, description=description)
```

Comment:
optparse has been deprecated for a long time now, but I know we already use it in quite a few places. Maybe worth using `argparse` for less future work? Or just as easy to do a single sweep over the whole repo if/when that becomes an issue. Not sure